pyro/autohelp/ast_visitor.py

```diff
@@ -0,0 +1,246 @@
+import enum
+from typing import Optional
+
+import libcst
+
+
+class Actions(enum.Enum):
+    # we don't care about any of these values, just using them as sentinels
+    DECORATOR_EVENT_CALLED = enum.auto()
+    DECORATOR_LISTEN_NOT_CALLED = enum.auto()
+    USING_SELF_ON_BOT_COMMAND = enum.auto()
+    CLIENT_IS_NOT_BOT = enum.auto()
+    PROCESS_COMMANDS_NOT_CALLED = enum.auto()
+
+
+class BaseHelpTransformer(libcst.CSTTransformer):
+    def __init__(self, find_all: bool = False):
+        super().__init__()
+        self.found_errors: list[Actions] = []
+        self.find_all = find_all
+        self.updates = {}
+
+    def _check_code(self) -> bool:
+        return bool(self.found_errors)
+
+    def on_visit(self, node):
+        if self.found_errors and not self.find_all:
+            # don't visit anything else
+            return False
+        res = super().on_visit(node)
+        if self.found_errors and not self.find_all:
+            # don't visit anything else
+            return False
+        return res
+
+    def on_leave(self, original_node, updated_node):
+        if changes := self.updates.get(original_node):
+            return changes(updated_node)
+        return updated_node
+
+
+class _FindCommandDecorator(libcst.CSTVisitor):
+    def __init__(self):
+        super().__init__()
+        self.is_command_dec = False
+
+    def visit_Decorator(self, node):
+        if not isinstance(node.decorator, libcst.Call):
+            return True
+        func = node.decorator.func
+        # find decorators that are like @thing.slash_command
+        # but ensure to ignore @commands.command` since that should have self.`
+        if func.attr.value.endswith("command") and func.value not in (
+            "commands",
+            "nextcord",
```

Comment:
Add disnake

Reply:
nope. Disnake does not have a `disnake.command` decorator.

Reply:
Oki